app/scripts/lib/validate.js

```diff
@@ -103,6 +103,47 @@ define(function (require, exports, module) {
       }
 
       return password.length >= Constants.PASSWORD_MIN_LENGTH;
+    },
+
+    /**
+     * Check whether data matches schema, depth-first.
+     * Compares using Object.prototype.toString.
+     */
+    isDataValid: function isDataValid (data, schema) {
+      if (! schema) {
+        return false;
+      }
+
+      if (typeof schema === 'string') {
+        if (schema[0] === '?') {
+          if (data === null || data === undefined) {
+            return true;
+          }
+
+          schema = schema.substr(1);
+        }
+
+        return Object.prototype.toString.call(data) === '[object ' + schema + ']';
```

Comment:
I can deal with that. The end result should be the same, one would have thrown an exception, this variant should return false if the schema is incorrect.